tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/xml/XmlParserHelper.java

```diff
@@ -120,7 +121,10 @@ public <T> T getJaxbRoot(InputStream stream, Class<T> clazz, Schema schema)
 		ContextProvidingValidationEventHandler handler = new ContextProvidingValidationEventHandler();
 		try {
 			staxEventReader = new JpaNamespaceTransformingEventReader( staxEventReader );
-			JAXBContext jaxbContext = JAXBContext.newInstance( ObjectFactory.class );
+			ClassLoader cl = this.getClass().getClassLoader();
+			String packageName = Optional.ofNullable(ObjectFactory.class.getPackage())
+					.map(p -> p.getName()).orElse("org.hibernate.processor.xml.jaxb");
+			JAXBContext jaxbContext = cl == null ? JAXBContext.newInstance( packageName ) : JAXBContext.newInstance( packageName, cl );
```

Comment:
Under what conditions does `this.getClass().getClassLoader()` return `null`?

At least there should be a comment explaining when this happens.

Reply:
I shouldn't but same checkstyle thing as above : since it could return null, I get error violation too 😞 

If using this syntax
```java
JAXBContext jaxbContext = JAXBContext.newInstance( packageName, this.getClass().getClassLoader() );
```

I get
```
.../hibernate-orm/tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/xml/XmlParserHelper.java:126: error: [argument] incompatible argument for parameter classLoader of JAXBContext.newInstance.
                        JAXBContext jaxbContext = JAXBContext.newInstance( packageName, this.getClass().getClassLoader() );
                                                                                                                      ^
  found   : @Initialized @Nullable ClassLoader
  required: @Initialized @NonNull ClassLoader
```